src/native-common/RootView.tsx

```diff
@@ -115,14 +126,22 @@ export class RootView extends React.Component<{}, RootViewState> {
     private _getStateFromStore(): RootViewState {
         let mainView = MainViewStore.getMainView();
 
-        if (mainView && this.props) {
-            mainView = React.cloneElement(mainView, this.props);
+        if (mainView && !_.isEqual(mainView.props, this._mainViewProps)) {
+            mainView = React.cloneElement(mainView, this._mainViewProps);
         }
 
         return {
-            mainView: mainView
+            mainView: mainView || this.state.mainView
```

Comment:
Do we want to allow `mainView` to be set to null via the store?  The reason I'm not allowing it here is to prevent the `componentWillMount` setState call overriding the mainView setup from the ctor.  If we do want to allow null from store, this can be refactored to prevent the overriding in a different manner.